frontend/src/components/LineChartPreview.tsx

```diff
@@ -19,12 +19,29 @@ type Props = {
   data?: Array<any>;
   summaryBelow: boolean;
   isPreview?: boolean;
+  colors?: {
+    primary: string | undefined;
+    secondary: string | undefined;
+  };
 };
 
 const LineChartPreview = (props: Props) => {
   const [linesHover, setLinesHover] = useState(null);
   const [hiddenLines, setHiddenLines] = useState<Array<string>>([]);
-  const colors = useColors(props.lines.length);
+  const [chartColors, setChartColors] = useState<Array<string>>([]);
+  const { settings } = useSettings();
+
+  useEffect(() => {
+    if (settings && settings.colors) {
+      const colors = UtilsService.getColors(
+        props.lines.length,
+        props.colors?.primary || settings.colors.primary,
+        props.colors?.secondary || settings.colors.secondary
+      );
+      setChartColors(colors);
+    }
+  }, [settings, props.lines, props.colors]);
```

Comment:
potential opportunity to abstract into a custom hook. I see these lines repeated in the other Widgets. A separate PR later for the refactor is fine.

Reply:
Good idea